server/handler/base.go

```diff
@@ -111,6 +122,87 @@ func (b *Base) postGitHubRepoStatus(ctx context.Context, client *github.Client,
 	return err
 }
 
+func (b *Base) PostCheckResults(ctx context.Context, client *github.Client, pr *github.PullRequest, evaluationResult common.Result) error {
+	sha := pr.GetHead().GetSHA()
+	owner := pr.GetBase().GetRepo().GetOwner().GetLogin()
+	repo := pr.GetBase().GetRepo().GetName()
+
+	for _, result := range evaluationResult.Children {
+		if err := b.createGitHubRepoCheck(ctx, client, pr, owner, repo, sha, *result); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *Base) createGitHubRepoCheck(ctx context.Context, client *github.Client, pr *github.PullRequest, owner string, repo string, sha string, evaluationResult common.Result) error {
+	publicURL := strings.TrimSuffix(b.BaseConfig.PublicURL, "/")
+	detailsURL := fmt.Sprintf("%s/details/%s/%s/%d", publicURL, owner, repo, pr.GetNumber())
+	defaultStatus := "completed"
+	time := github.Timestamp{
+		Time: time.Now(),
+	}
+
+	conclusion := "neutral"
+	switch evaluationResult.Status.String() {
+	case "approved":
+		conclusion = "success"
+	case "skipped":
+		conclusion = "neutral"
+	case "disapproved":
+		conclusion = "failure"
+	case "unknown":
+		conclusion = "failure"
+	case "pending":
+		conclusion = "failure"
+	}
+
+	var builder strings.Builder
+	if len(evaluationResult.Children) == 0 {
+		_, _ = builder.WriteString(evaluationResult.Description)
+	}
+
+	childSummaries := b.generateCheckRunSummary(evaluationResult.Children, 0)
+	_, _ = builder.WriteString(childSummaries)
+	summary := builder.String()
+	checkName := fmt.Sprintf("%s: %ss - %s", b.PullOpts.StatusCheckContext, evaluationResult.Name, pr.GetBase().GetRef())
+
+	status := &github.CreateCheckRunOptions{
+		Name:       checkName,
+		DetailsURL: &detailsURL,
+		Status:     &defaultStatus,
+		Output: &github.CheckRunOutput{
+			Title:   &evaluationResult.Description,
+			Summary: &summary,
+		},
+		StartedAt:   &time,
+		CompletedAt: &time,
+		Conclusion:  &conclusion,
+		HeadSHA:     sha,
+	}
+
+	_, _, err := client.Checks.CreateCheckRun(ctx, owner, repo, *status)
+	return err
+}
+
+func (b *Base) generateCheckRunSummary(results []*common.Result, level int) string {
+	var summaryBuilder strings.Builder
+	headerValue := strings.Repeat("#", 3+level)
+	indentValue := strings.Repeat(" ", 2*level)
+
+	for _, childResult := range results {
+		_, _ = summaryBuilder.WriteString("\n")
+		_, _ = summaryBuilder.WriteString(fmt.Sprintf("%s- %s %s (%s)", indentValue, headerValue, strings.Title(childResult.Name), childResult.Status.String()))
+		_, _ = summaryBuilder.WriteString("\n")
+		_, _ = summaryBuilder.WriteString(fmt.Sprintf("  %s%s", indentValue, childResult.Description))
+		childSummaries := b.generateCheckRunSummary(childResult.Children, level+1)
+		_, _ = summaryBuilder.WriteString(childSummaries)
+	}
+
+	return summaryBuilder.String()
```

Comment:
Did you look at using `text/template` for this, similar to how the UI version is rendered? I'm not sure if it would be simpler, but a recursive template might be easier to understand than the string formatting indentation here.